pkg/cloudprovider/provider/hetzner/provider.go

```diff
@@ -183,6 +192,14 @@ func (p *provider) Validate(spec v1alpha1.MachineSpec) error {
 		}
 	}
 
+	if len(c.Firewalls) != 0 {
+		for _, firewall := range c.Firewalls {
+			if n, _, err := client.Firewall.Get(ctx, firewall); err != nil || n == nil {
+				return fmt.Errorf("failed to get firewall %q: %v", firewall, err)
+			}
```

Comment:
Let split this check into two `if`s. Right now, if the firewall doesn't exist, this is going to produce an error such as:

```
failed to get firewall "machine-controller-e2e": <nil>
```

From this error, it's hard to find out that the firewall doesn't exist. Instead, we can do this:

```suggestion
			if f, _, err := client.Firewall.Get(ctx, firewall); err != nil {
				return fmt.Errorf("failed to get firewall %q: %v", firewall, err)
			}
			if f == nil {
				return fmt.Errorf("firewall %q does not exist", firewall)
			}
```